### PR TITLE
Devcontainer to configure git and precommit

### PR DESCRIPTION
**Pull request by brucearctor**
## Summary

This change configures git to be able to run from within the devcontainer.  

It also 'installs' pre-commit.

I was following https://docs.astral.sh/ruff/contributing/#prerequisites and realized that it is better to just build on top of current devcontainer infra that is already supported by the community, and have even fewer manual steps.

## Test Plan

This was tested by running locally.

### CODE REVIEW

File: .devcontainer/post-create.sh [hunk 1]
```diff
@@ -6,3 +6,9 @@ cargo install cargo-insta
 cargo fetch
 
 pip install maturin pre-commit
+
+# configure git to be able to run from the devcontainer
+git config --global --add safe.directory /workspaces/ruff
+
+# ensure pre-commit is configured to run
+pre-commit install
```
Comment (dhruvmanila) [inline .devcontainer/post-create.sh:14]: I'm a bit hesitant to enforce pre-commit to contributors inside the dev containers or more generally I'm hesitant to install anything other than the required tools to be able to contribute to Ruff. I, personally, don't install the hooks and run them manually as I might commit a lot.

File: .devcontainer/post-create.sh [hunk 2]
```diff
@@ -6,3 +6,9 @@ cargo install cargo-insta
 cargo fetch
 
 pip install maturin pre-commit
+
+# configure git to be able to run from the devcontainer
+git config --global --add safe.directory /workspaces/ruff
```
Comment (dhruvmanila) [inline .devcontainer/post-create.sh:11]: Is there a reference to this requirement?

File: .devcontainer/post-create.sh (hunk 1 above)
Comment (brucearctor) [reply .devcontainer/post-create.sh:14]: i guess it ultimately is comes down to the prevailing preference.  I generally prefer about anything I might want to be installed, and I can always disable.  BUT, I understand the alternative, so can close this PR.

Comment (dhruvmanila) [reply .devcontainer/post-create.sh:14]: Thank you for understanding :)

**Pull request closed without merging**